development/development/php/character_set_and_encodings.rst

```diff
@@ -0,0 +1,57 @@
+5. Character Sets and Encodings
+===============================
+
+What are Unicode, UCS and UTF-8?
+--------------------------------
+
+The `Universal Character Set (UCS) <http://en.wikipedia.org/wiki/Universal_Character_Set>`_ described in ISO/IEC 10646 consists of a large amount of characters. Each of them has a unique name and a code point which is an integer number. `Unicode <http://en.wikipedia.org/wiki/Unicode>`_ - which is an industry standard - complements the Universal Character Set with further information about the characters' properties and alternative character encodings. More information on Unicode can be found on the `Unicode Consortium's website <http://www.unicode.org/>`_. One of the Unicode encodings is the `8-bit Unicode Transformation Format (UTF-8) <http://en.wikipedia.org/wiki/UTF-8>`_. It encodes characters with up to four bytes aiming for maximum compatibility with the `American Standard Code for Information Interchange <http://en.wikipedia.org/wiki/ASCII>`_ which is a 7-bit encoding of a relatively small subset of the UCS.
+
+phpBB's use of Unicode
+++++++++++++++++++++++
+
+Unfortunately PHP does not facilitate the use of Unicode prior to version 6. Most functions simply treat strings as sequences of bytes assuming that each character takes up exactly one byte. This behaviour still allows for storing UTF-8 encoded text in PHP strings but many operations on strings have unexpected results. To circumvent this problem we have created some alternative functions to PHP's native string operations which use code points instead of bytes. These functions can be found in ``/includes/utf/utf_tools.php``. A lot of native PHP functions still work with UTF-8 as long as you stick to certain restrictions. For example ``explode`` still works as long as the first and the last character of the delimiter string are ASCII characters.
```

Comment:
Are you sure about version 6?

Reply:
Whatever you recommend here I just copied it directly from the wiki

Reply:
Well ok, let's just leave it as is (easter egg style)